pkg/scaling/executor/scale_jobs.go

```diff
@@ -126,3 +132,89 @@ func (e *scaleExecutor) getRunningJobCount(scaledJob *kedav1alpha1.ScaledJob, ma
 
 	return runningJobs
 }
+
+// Clean up will delete the jobs that is exceed historyLimit
+func (e *scaleExecutor) cleanUp(scaledJob *kedav1alpha1.ScaledJob) error {
+	opts := []client.ListOption{
+		client.InNamespace(scaledJob.GetNamespace()),
+		client.MatchingLabels(map[string]string{"scaledjob": scaledJob.GetName()}),
+	}
+
+	jobs := &batchv1.JobList{}
+	err := e.client.List(context.TODO(), jobs, opts...)
+	if err != nil {
+		e.logger.Info("Can not get job list: ", scaledJob.GetName())
+		return err
+	}
+
+	completedJobs := []batchv1.Job{}
+	failedJobs := []batchv1.Job{}
+	for _, job := range jobs.Items {
+		finishedJobConditionType := e.getFinishedJobConditionType(&job)
+		if finishedJobConditionType == batchv1.JobComplete {
+			completedJobs = append(completedJobs, job)
+			continue
+		}
+		if finishedJobConditionType == batchv1.JobFailed {
+			failedJobs = append(failedJobs, job)
+		}
```

Comment:
maybe a switch case would be more readable here?

Reply:
Agree! Done